apache2/apache2_util.c

```diff
@@ -289,24 +290,24 @@ static void get_ruleset_type_version(char* waf_ruleset_info, char* waf_ruleset_t
     }
 }
 
-static int write_file_with_lock(apr_global_mutex_t* lock, apr_file_t* fd, char* str) {
+static int write_file_with_lock(struct waf_lock* lock, apr_file_t* fd, char* str) {
```

Comment:
I don't know whether we need to keep the original code or not and whether we can directly change the API. Besides, If this code is only for the JSON log, could we separate it into one independent commit?

Reply:
I think since it's a fix for the bug, we don't really need to keep the original code, will separate into another pr